### PR TITLE
chore: add grafana faro sdk (CE)

### PR DESCRIPTION
**Pull request by dvj1988**
## Description
- Remove new relic browser agent
- Add faro sdk to capture frontend perf metrics and traces.

Fixes #`Issue Number`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/12490844984>
> Commit: c9d4264027467bf33e1de519eb69c7762b6e7f75
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=12490844984&attempt=2" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.All`
> Spec:
> <hr>Wed, 25 Dec 2024 09:33:26 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Introduced new environment variable `APPSMITH_HOSTNAME` for dynamic hostname configuration in HTML files.
	- Enhanced telemetry capabilities with new imports and updated types for better observability.
	- Added `tracingUrl` under the observability section in configuration files for improved telemetry tracking.

- **Bug Fixes**
	- Adjusted telemetry data handling to utilize new `Attributes` type for improved consistency.

- **Documentation**
	- Updated import paths for various telemetry-related components to reflect new module organization.

- **Chores**
	- Removed deprecated telemetry configurations and streamlined build processes.
	- Updated Nginx configuration to reflect new telemetry parameters.

### CODE REVIEW

**Review by subhrashisdas — DISMISSED** [review]

File: app/client/src/plugins/Linting/types.ts [hunk 1]
```diff
@@ -11,11 +11,11 @@ import type {
 import type { DependencyMap } from "utils/DynamicBindingUtils";
 import type { TJSPropertiesState } from "workers/Evaluation/JSObject/jsPropertiesState";
 import type { JSLibrary } from "workers/common/JSLibrary";
-import type { WebworkerSpanData } from "UITelemetry/generateWebWorkerTraces";
-import type { SpanAttributes } from "UITelemetry/generateTraces";
+import type { WebworkerSpanData } from "instrumentation/types";
 import type { LINTER_TYPE } from "./constants";
+import type { Attributes } from "@opentelemetry/api";
```
Comment (rajatagrawal) [inline app/client/src/plugins/Linting/types.ts:16]: One theoretical concern with importing opentelemetry api directly across the codebase is that it introduces tight coupling between our code and the instrumentation library. It prevents us to switch between open telemetry and zipkin for example by just changing the library import and needs replacement everywhere in the code. What do you think ?

Comment (dvj1988) [reply app/client/src/plugins/Linting/types.ts:16]: Understood, we can move this to a common type in instrumentation/types.ts

Comment (dvj1988) [reply app/client/src/plugins/Linting/types.ts]: @rajatagrawal Resolved

File: app/client/knip.json [hunk 2]
```diff
@@ -35,7 +35,6 @@
     "postcss-url",
     "@types/dom-view-transitions",
     "@types/jest",
-    "@sentry/webpack-plugin",
```
Comment (KelvinOm) [inline app/client/knip.json:38]: "@sentry/react": "^6.2.4" — should this package also be deleted?

File: app/client/public/index.html [hunk 3]
```diff
@@ -214,19 +214,8 @@
...
-          accountId: parseConfig('{{env "APPSMITH_NEW_RELIC_ACCOUNT_ID"}}'),
-          applicationId: parseConfig('{{env "APPSMITH_NEW_RELIC_APPLICATION_ID"}}'),
-          browserAgentlicenseKey: parseConfig('{{env "APPSMITH_NEW_RELIC_BROWSER_AGENT_LICENSE_KEY"}}'),
-          browserAgentEndpoint: parseConfig('{{env "APPSMITH_NEW_RELIC_BROWSER_AGENT_ENDPOINT"}}'),
-          otlpLicenseKey: parseConfig('{{env "APPSMITH_NEW_RELIC_OTLP_LICENSE_KEY"}}'),
-          //OTLP following the naming convention of Sdk initialisation
-          otlpServiceName: parseConfig('{{env "APPSMITH_NEW_RELIC_OTEL_SERVICE_NAME"}}'),
-          otlpEndpoint:parseConfig('{{env "APPSMITH_NEW_RELIC_OTEL_EXPORTER_OTLP_ENDPOINT"}}'),
-
+          serviceInstanceId: parseConfig('{{env "APPSMITH_HOSTNAME"}}') || "appsmith-0",
```
Comment (rajatagrawal) [inline app/client/public/index.html:217]: Do the default values need to be defined here or can they be defined in the function that uses this environment variable to give final value for this property ?

File: app/client/knip.json (hunk 2 above)
Comment (dvj1988) [reply app/client/knip.json:38]: We are planning to do this in the next pr.

File: app/client/public/index.html (hunk 3 above)
Comment (dvj1988) [reply app/client/public/index.html:217]: We are doing it in both the places now. I think we can remove it from here.

**Review by rajatagrawal — APPROVED** [review]

File: app/client/src/widgets/withWidgetProps.tsx [hunk 4]
```diff
@@ -371,7 +374,7 @@ function withWidgetProps(WrappedWidget: typeof BaseWidget) {
       }
     }
 
-    if (!newRelic.enableNewRelic) {
+    if (!tracingUrl) {
```
Comment (rajatagrawal) [inline app/client/src/widgets/withWidgetProps.tsx:377]: nit: Is there merit in wrapping this logic into a fn : tracingEnabled() to improve readability ?

Comment (dvj1988) [reply app/client/src/widgets/withWidgetProps.tsx:377]: yeah makes sense. Will add this.

File: app/client/public/index.html (hunk 3 above)
Comment (dvj1988) [reply app/client/public/index.html]: Resolved.

File: app/client/src/widgets/withWidgetProps.tsx (hunk 4 above)
Comment (dvj1988) [reply app/client/src/widgets/withWidgetProps.tsx]: resolved.

**Review by rajatagrawal — APPROVED** [review]

**Comment by Yiidiir** [comment]
@dvj1988  I'm running Appsmith CE self-hosted and started seeing queries going to a grafana endpoint, for compliance reasons I have to disable this, is there an environment variable to switch this off.. Share anonymous data in settings in unchecked and `APPSMITH_DISABLE_TELEMETRY` is set to true but for some reason is not respected for grafana faro.

@rajatagrawal @dvj1988 please advise.

**Comment by dvj1988** [comment]
> @dvj1988  I'm running Appsmith CE self-hosted and started seeing queries going to a grafana endpoint, for compliance reasons I have to disable this, is there an environment variable to switch this off.. Share anonymous data in settings in unchecked and `APPSMITH_DISABLE_TELEMETRY` is set to true but for some reason is not respected for grafana faro.
> 
> 
> 
> @rajatagrawal @dvj1988 please advise.

Hello Yidir. Could you share the version of Appsmith you are using?

**Comment by Yiidiir** [comment]
> > @dvj1988  I'm running Appsmith CE self-hosted and started seeing queries going to a grafana endpoint, for compliance reasons I have to disable this, is there an environment variable to switch this off.. Share anonymous data in settings in unchecked and `APPSMITH_DISABLE_TELEMETRY` is set to true but for some reason is not respected for grafana faro.
> > @rajatagrawal @dvj1988 please advise.
> 
> Hello Yiidiir. Could you share the version of Appsmith you are using?

Appsmith Community v1.55

**Comment by dvj1988** [comment]
@Yiidiir The faro telemetry is not enabled by default on community editions. We have verified this on the community edition you have mentioned.

Can you send us more details on the network calls that you see from the frontend client? Screengrabs or screenshots of your network panel with request details to the faro calls would help.

**Comment by dvj1988** [comment]
@Yiidiir Thank you for bringing this to our attention. The telemetry you observed is associated with the AI assistant in the custom widget builder. Please note that we are only collecting page load performance metrics for the AI assistant, and no personal information is included in this telemetry.

Rest assured, we will be disabling telemetry for the AI assistant in an upcoming version.
cc: @nikhil @rajatagrawal

**Comment by Yiidiir** [comment]
Thank you @dvj1988 
Talking about the AI assistant, can we use our own openai API key?

**Pull request merged**